### PR TITLE
JDK-8326140: src/jdk.accessibility/windows/native/libjavaaccessbridge/AccessBridgeJavaEntryPoints.cpp ReleaseStringChars might be missing in early returns

### PR DESCRIPTION
**Pull request by MBaesken**
In AccessBridgeJavaEntryPoints.cpp we have a couple of exception checks with potential early returns. Those miss ReleaseStringChars .

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8326140](https://bugs.openjdk.org/browse/JDK-8326140): src/jdk.accessibility/windows/native/libjavaaccessbridge/AccessBridgeJavaEntryPoints.cpp ReleaseStringChars might be missing in early returns (**Bug** - P4)

### Reviewers
 * [Christoph Langer](https://openjdk.org/census#clanger) (@RealCLanger - **Reviewer**) ⚠️ Review applies to [4beaad99](https://git.openjdk.org/jdk/pull/17915/files/4beaad991e38ebbba86cfb2ddf142876764f49ae)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17915/head:pull/17915` \
`$ git checkout pull/17915`

Update a local copy of the PR: \
`$ git checkout pull/17915` \
`$ git pull https://git.openjdk.org/jdk.git pull/17915/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17915`

View PR using the GUI difftool: \
`$ git pr show -t 17915`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17915.diff">https://git.openjdk.org/jdk/pull/17915.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17915#issuecomment-1952423339)

### CODE REVIEW

**Comment by MBaesken** [comment]
btw there might be further room for improvement / simplification. The current macro code blocks look like this

```
            EXCEPTION_CHECK_WITH_RELEASE("message1", FALSE, js, stringBytes);
            EXCEPTION_CHECK("message2 for ReleaseStringChars()", FALSE);
```

The second parameter could be omitted because it is always FALSE anyway it those EXCEPTION_CHECK_WITH_RELEASE blocks.
The second line EXCEPTION_CHECK could be moved into the macro EXCEPTION_CHECK_WITH_RELEASE  itself .

**Review by RealCLanger — APPROVED** [review]
LGTM

File: src/jdk.accessibility/windows/native/libjavaaccessbridge/AccessBridgeJavaEntryPoints.cpp [hunk 1]
```diff
@@ -1735,11 +1743,10 @@ AccessBridgeJavaEntryPoints::getAccessibleContextInfo(jobject accessibleContext,
             EXCEPTION_CHECK("Getting AccessibleName - call to GetStringChars()", FALSE);
             wcsncpy(info->name, stringBytes, (sizeof(info->name) / sizeof(wchar_t)));
             length = jniEnv->GetStringLength(js);
+            EXCEPTION_CHECK_WITH_RELEASE("Getting AccessibleName - call to GetStringLength()", FALSE, js, stringBytes);
+            EXCEPTION_CHECK("Getting AccessibleName - call to ReleaseStringChars()", FALSE);
```
Comment (prrace) [inline src/jdk.accessibility/windows/native/libjavaaccessbridge/AccessBridgeJavaEntryPoints.cpp:1747]: So this is for the call to ReleaseStringChars() when there is no exception doing the GetStringLength
Since all the calls to ReleaseStringChars are now moved inside the WITH_RELEASE macro, it would seem more
logical to me to move this 2nd macro call inside the definition of the new macro (option 1)
Either that OR  (option 1) do not move the "no previous exception" case inside the new macro at all
In other words, this CHECK should be placed next to where I can see the call that requires it.
Now, you'll probably tell me it isn't needed in the cases where the method immediately returns, so if that causes
a problem then option 2 seems like it would be better.

**Comment by MBaesken** [comment]
> In other words, this CHECK should be placed next to where I can see the call that requires it.

Hi Phil, I thought about this too. Just one EXCEPTION_CHECK_WITH_RELEASE macro and there the second check inside this macro, the two   EXCEPTION_CHECK_WITH_RELEASE and EXCEPTION_CHECK just after it look a bit odd.
Should I change it this way and use just one macro ?  Btw. at some places the  ` EXCEPTION_CHECK_WITH_RELEASE` is placed not directly placed after the` jniEnv->GetStringLength`  call ; should I move it up there or keep it where it was ?

**Comment by MBaesken** [comment]
> > In other words, this CHECK should be placed next to where I can see the call that requires it.
> 
> Hi Phil, I thought about this too. Just one EXCEPTION_CHECK_WITH_RELEASE macro and there the second check inside this macro, the two EXCEPTION_CHECK_WITH_RELEASE and EXCEPTION_CHECK just after it look a bit odd. 


Adjusted the coding, now do more in the EXCEPTION_CHECK_WITH_RELEASE macro. Please check .

**Comment by MBaesken** [comment]
Hi Christoph and Phil, thanks for the reviews !

/integrate
